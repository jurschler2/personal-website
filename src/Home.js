import React from "react";
// import {images} from "./data";


function Home() {

  return (
    <>
      {/* <div style={{backgroundImage: `url(${images.background.source})`, height: '100%', width: '100%'}}>Meet Joe Urschler</div> */}
      <div>Home</div>
      {/* <img src={images.background.source} alt="" height='100%' width='100%'/> */}

    </>
  )
}

export default Home;