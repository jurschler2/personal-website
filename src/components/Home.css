/** 
 * Styling for the Home component:
 */

.intro-hello {
  font-family: Kanit;
  font-weight: 300;
  font-size: 150%;
  color: #7b7b87;
  display:  block;
  margin: -1%;
}
.intro-name {
  font-weight: 400;
  font-size: 400%;
  color: #39384b;
  display:  block;
  margin: -1%;
}
.intro-tagline {
  font-weight: 400;
  font-size: 400%;
  color: #7b7b87;
  display:  block;
  margin: -1%;
}

.home-container {
  margin-left: 30%;
  margin-top: 5%;
  text-align: left;
  padding: 2%;
}