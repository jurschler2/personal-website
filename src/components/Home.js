import React from "react";
import { Container, Row, Col } from "react-bootstrap"
// import "./Home.css"

/**
 *  DESCRIPTION:
 *  PROPS: None
 *  FLOW: App => Routes => Home
 *  PARENT: Routes
 *  CHILDREN:
 */

function Home() {

  return (
    <Container className="sectionContainer" id="/">
      <Row>
        <Col xs={{ span: 12, offset: 0 }} md={{ span: 12, offset: 0 }}>
          <h4 data-aos-delay="100" data-aos="fade"className="intro-hello">Hello there, my name is</h4>
          <h2 data-aos-delay="900" data-aos="fade" className="intro-name">Joseph Urschler</h2>
          <div data-aos-delay="1500" data-aos="fade-right"className="homeRule"></div>
          <h3 data-aos-delay="2000" data-aos="fade" className="intro-tagline">I like to build things.</h3>
        </Col>
      </Row>
    </Container>
  )
}

export default Home;