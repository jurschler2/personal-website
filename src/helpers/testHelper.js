import React from "react";
import { render } from "@testing-library/react";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";

export function renderWithRouter(children, initialPath = "") {
  const history = createMemoryHistory();
  if (initialPath) history.push(initialPath);
  return render(<Router history={history}>{children}</Router>);
}
