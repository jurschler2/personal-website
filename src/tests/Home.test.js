import React from "react";
import Home from "../components/Home";
import {renderWithRouter} from "../helpers/testHelper"

it("renders without crashing", () => {
  renderWithRouter(<Home />);
});

it("matches snapshot", () => {
  const { asFragment } = renderWithRouter(<Home />);
  expect(asFragment()).toMatchSnapshot();
});