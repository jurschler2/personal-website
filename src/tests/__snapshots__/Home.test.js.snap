// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="home-page"
  >
    <div
      class="home-container"
    >
      <p
        class="intro-hello"
      >
        Hello there, my name is
      </p>
      <p
        class="intro-name"
      >
        Joseph Urschler.
      </p>
      <p
        class="intro-tagline"
      >
        I like to build things.
      </p>
    </div>
  </div>
</DocumentFragment>
`;
